### DISCUSSION TITLE
why-you-dont-need-has-and-belongs-to-many

### DISCUSSION THREAD

**Discussion by Em-Arce** [avionschool/batch8-activities#20] [Ideas]
TLDR: 
has many and belongs to: roles_users table has no primary key (no id), only stores user_id and role_id 

has many throught: role_users table has primary key (id) in addition to user_id and role_id

Better to anticipate that you will need additional attributes from the join table since it will be easier and faster to do migrations (just add/change column etc) than to undo the has many and belongs to then set up the has many through association.

Source: https://flatironschool.com/blog/why-you-dont-need-has-and-belongs-to-many

**Comment by lea-tinoso** [comment]
@yvettetan HYG.
